### DISCUSSION TITLE
RP2040 as an audio USB device?

### DISCUSSION THREAD

**Discussion by yuri-evangelista** [adafruit/Adafruit_TinyUSB_Arduino#348] [General]
Hi everyone, noob developer here!

I'm struggling trying to create a USB stereo mic. I'm using two I2S mics (INMP441) and the Earle F. Philhower arduino-pico core.
Is there a way for getting the board be recognized as an USB audio device which streams the mics data? 
Right now I can only stream the raw data via serial, which is almost unuseful...

Thank you very much
Yuri

**Comment by pschatzmann** [comment]
I managed to implement a first DRAFT of an Audio class (implementing a Speaker and Microphone): details can be found in the wiki of [my fork
](https://github.com/pschatzmann/Adafruit_TinyUSB_Arduino/tree/Audio)

**Reply by h5n1xp** [reply]
I've just managed to get my Raspberry PI Pico working as an audio class device using @pschatzmann 's fork of TinyUSB, does anyone know why he main branch doesn't implement the class yet?

**Reply by pschatzmann** [reply]
- The major reason is that this is not working with ESP32 and I am not sure how to address this issue 
- I did only some crude testing on other platforms
- I am waiting for the driver correction which is available in TinyUSB to make it up to this project
